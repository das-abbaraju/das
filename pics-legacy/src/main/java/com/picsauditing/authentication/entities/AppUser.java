package com.picsauditing.authentication.entities;

import org.hibernate.annotations.Cache;
import org.hibernate.annotations.CacheConcurrencyStrategy;

import javax.persistence.*;
import javax.persistence.Entity;
import javax.persistence.Table;
import java.io.Serializable;

@Entity
@Table(name="app_user")
@Cache(usage = CacheConcurrencyStrategy.READ_WRITE, region = "daily")
public class AppUser implements Serializable {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;

	private String username;
	private String password = "";
	private String hashSalt;
	private String resetHash;

	public AppUser() {
	}

	public AppUser(int id) {
		this.id = id;
	}

	public AppUser(String username, String password) {
		this.username = username;
		this.password = password;
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getUsername() {
		return username;
	}

	public void setUsername(String username) {
		this.username = username;
	}

	public String getPassword() {
		return password;
	}

	public void setPassword(String password) {
		this.password = password;
	}

	public String getHashSalt() {
		return hashSalt;
	}

	public void setHashSalt(String hashSalt) {
		this.hashSalt = hashSalt;
	}

	public String getResetHash() {
		return resetHash;
	}

	public void setResetHash(String resetHash) {
		this.resetHash = resetHash;
	}
}
