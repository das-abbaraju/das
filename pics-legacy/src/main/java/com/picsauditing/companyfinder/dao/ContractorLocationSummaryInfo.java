package com.picsauditing.companyfinder.dao;

public class ContractorLocationSummaryInfo {
    private int id;
    private int conId;
    private double latitude;
    private double longitude;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getConId() {
        return conId;
    }

    public void setConId(int conId) {
        this.conId = conId;
    }

    public double getLatitude() {
        return latitude;
    }

    public void setLatitude(double latitude) {
        this.latitude = latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public void setLongitude(double longitude) {
        this.longitude = longitude;
    }
}
