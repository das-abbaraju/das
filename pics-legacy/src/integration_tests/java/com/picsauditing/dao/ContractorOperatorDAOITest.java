package com.picsauditing.dao;

import java.util.HashSet;
import java.util.List;
import java.util.Set;

import javax.sql.DataSource;

import com.picsauditing.jpa.entities.ContractorAccount;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;
import org.powermock.reflect.Whitebox;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;

import com.picsauditing.PICS.I18nCache;
import com.picsauditing.jpa.entities.Account;
import com.picsauditing.jpa.entities.ContractorOperator;
import com.picsauditing.search.Database;

import static org.junit.Assert.assertTrue;

@RunWith(SpringJUnit4ClassRunner.class)
@ContextConfiguration(locations={"ContractorOperatorDAOITest-context.xml"})
public class ContractorOperatorDAOITest {
	private JdbcTemplate jdbcTemplate;
	
	@Autowired private ContractorOperatorDAO contractorOperatorDAO;
	@Autowired private ContractorAccountDAO contractorAccountDAO;

	@Mock private Database databaseForTesting;
	
	@Autowired 
    public void setDataSource(DataSource dataSource) {
        this.jdbcTemplate = new JdbcTemplate(dataSource);
    }
	
	@Before
	public void setUp() throws Exception {
		MockitoAnnotations.initMocks(this);
		Whitebox.setInternalState(I18nCache.class, "databaseForTesting", databaseForTesting);
	}

    @Test
    public void testCA() {
        ContractorAccount contractorAccount = contractorAccountDAO.find(39202);
        assertTrue(true);
    }

	@Test
	public void testFindForOperators() {
		@SuppressWarnings({ "rawtypes", "unchecked", "serial" })
		Set<Integer> operatorIds = new HashSet() {{
			add(16); add(1068); add(1251); 
		}};
		Set<ContractorOperator> results = contractorOperatorDAO.findForOperators(357, operatorIds);
	}
}
