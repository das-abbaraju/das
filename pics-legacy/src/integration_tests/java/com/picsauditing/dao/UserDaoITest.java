package com.picsauditing.dao;

import static org.junit.Assert.*;
import static org.hamcrest.Matchers.*;

import com.picsauditing.PICS.I18nCache;
import com.picsauditing.jpa.entities.User;
import com.picsauditing.search.Database;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;
import org.powermock.reflect.Whitebox;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;

@RunWith(SpringJUnit4ClassRunner.class)
@ContextConfiguration(locations={"UserDAOITest-context.xml"})
public class UserDaoITest {
    @Mock
    private Database databaseForTesting;
    @Autowired
    private UserDAO userDAO;

    @Before
    public void setUp() throws Exception {
        MockitoAnnotations.initMocks(this);
        Whitebox.setInternalState(I18nCache.class, "databaseForTesting", databaseForTesting);
    }


    @Test
    public void testFindName() throws Exception {
        User user = userDAO.findName("gmeurer");
        assertThat(user, is(notNullValue()));
    }
}
