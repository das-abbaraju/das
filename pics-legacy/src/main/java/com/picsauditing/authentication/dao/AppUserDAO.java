package com.picsauditing.authentication.dao;

import com.picsauditing.authentication.entities.AppUser;

import javax.persistence.EntityManager;
import javax.persistence.PersistenceContext;
import javax.persistence.Query;
import javax.persistence.TypedQuery;
import java.util.List;

public class AppUserDAO {

	@PersistenceContext
	private EntityManager em;

	public AppUser save(AppUser o) {
		if (o.getId() == 0) {
			em.persist(o);
		} else {
			o = em.merge(o);
		}

		em.flush();

		return o;
	}

	public AppUser findById(final int id) {
		return em.find(AppUser.class, id);
	}

	public List<AppUser> findListByUserName(final String username) {
		TypedQuery<AppUser> query = em.createQuery("FROM AppUser " +
				"WHERE username = :username", AppUser.class);

		query.setParameter("username", username);

		return query.getResultList();
	}

	public AppUser findByUserName(String username) {
		TypedQuery<AppUser> q = em.createQuery("SELECT a FROM AppUser a " +
				"WHERE a.username = :username", AppUser.class);

		q.setParameter("username", username);

		try {
			return q.getSingleResult();
		} catch (Exception e) {
			return null;
		}
	}

	public AppUser findByUserNameAndPassword(final String username, final String password) {
		TypedQuery<AppUser> q = em.createQuery("SELECT a FROM AppUser a WHERE a.username = :username " +
				"AND a.password = :password", AppUser.class);

		q.setParameter("username", username);
		q.setParameter("password", password);

		try {
			return q.getSingleResult();
		} catch (Exception e) {
			return null;
		}
	}

	public boolean duplicateUsername(final String username, final int appUserID) {
		try {
			AppUser appUser = findByUserName(username);
			if (appUser == null) {
				return false;
			}

			int id = appUser.getId();
			if (id > 0) {
				// found a user with this username
				if (id != appUserID)
					// This is in use by another user
					return true;
			}
		} catch (Exception e) {
			//TODO
		}

		return false;
	}
}
