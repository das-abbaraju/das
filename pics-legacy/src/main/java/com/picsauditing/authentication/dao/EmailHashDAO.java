package com.picsauditing.authentication.dao;

import com.picsauditing.employeeguard.entities.EmailHash;
import org.springframework.transaction.annotation.Propagation;
import org.springframework.transaction.annotation.Transactional;

import javax.persistence.EntityManager;
import javax.persistence.PersistenceContext;
import javax.persistence.TypedQuery;

public class EmailHashDAO {

	@PersistenceContext
	protected EntityManager em;

	public EmailHash find(final int id) {
		return em.find(EmailHash.class, id);
	}

	public EmailHash findByHash(final String hash) {
		TypedQuery<EmailHash> query = em.createQuery("FROM EmailHash eh " +
				"WHERE eh.hash = :hash", EmailHash.class);

		query.setParameter("hash", hash);

		try {
			return query.getSingleResult();
		} catch (Exception e) {
			return null;
		}
	}

	@Transactional(propagation = Propagation.NESTED)
	public EmailHash save(EmailHash o) {
		if (o.getId() == 0) {
			em.persist(o);
		} else {
			o = em.merge(o);
		}

		em.flush();

		return o;
	}
}
