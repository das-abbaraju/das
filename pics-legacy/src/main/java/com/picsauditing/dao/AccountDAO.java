package com.picsauditing.dao;

import com.picsauditing.access.Permissions;
import com.picsauditing.jpa.entities.Account;
import com.picsauditing.jpa.entities.ContractorAccount;
import com.picsauditing.jpa.entities.OperatorAccount;
import com.picsauditing.util.Strings;
import org.apache.commons.collections.CollectionUtils;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import javax.persistence.Query;
import javax.persistence.TypedQuery;
import java.util.Collection;
import java.util.Collections;
import java.util.List;

@SuppressWarnings("unchecked")
public class AccountDAO extends PicsDAO {

    private static final Logger LOG = LoggerFactory.getLogger(AccountDAO.class);

	public Account find(int id) {
		Account a = em.find(Account.class, id);
		return a;
	}

	public Account find(int id, String type) {
		if ("Contractor".equals(type))
			return em.find(ContractorAccount.class, id);
		
		if ("Operator".equals(type) || "Corporate".equals(type))
			return em.find(OperatorAccount.class, id);
		
		return em.find(Account.class, id);
	}

	public List<Account> findWhere(String where) {
		if (where == null)
			where = "";
		if (where.length() > 0)
			where = "WHERE " + where;
		Query query = em.createQuery("select a from Account a " + where + " order by a.name");
		return query.getResultList();
	}
	
	public Account findRandomAssessmentCenter() {
		Query query = em.createQuery("SELECT a FROM Account a WHERE type = 'Assessment' ORDER BY RAND()");
		query.setMaxResults(1);
		
		return (Account) query.getSingleResult();
	}
	
	public List<Account> findNoteRestrictionOperators(Permissions permissions) {
		String where = "AND a.status IN ('Active'";
		boolean needsClosingParen = true;
		
		if (permissions.isAdmin() || permissions.getAccountStatus().isDemo())
			where += ",'Demo'";

		if (permissions.isAdmin() || permissions.isAuditor())
			where += ",'Pending'";
		if (permissions.isOperatorCorporate()) {
			needsClosingParen = false;
			where += ",'Pending') AND a.id IN (" + Strings.implode(permissions.getVisibleAccounts(), ",") + ")";
		}
		if (needsClosingParen)
			where += ")";

		Query query;

		query = em.createQuery("FROM Account a WHERE type IN ('Admin','Corporate', 'Operator') AND a.id!=10403 " + where + " ORDER BY a.name");

		return query.getResultList();
	}
	
	public int findByID(String username) {
		Query query = em.createQuery("SELECT id FROM Account WHERE username='?'");
		query.setParameter(1, Strings.escapeQuotesAndSlashes(username));
		return Integer.parseInt(query.getSingleResult().toString());
	}
	
	public List<Account> findSetWhere(String where, int limit) {
		if (where == null)
			where = "";
		if (where.length() > 0)
			where = "WHERE " + where;
		
		Query query = em.createQuery("FROM Account a " + where + " ORDER BY a.id");
		query.setMaxResults(limit);
		return query.getResultList();
	}

	public List<Account> findByIds(Collection<Integer> accountIds) {
        if (CollectionUtils.isEmpty(accountIds)) {
            return Collections.emptyList();
        }

        TypedQuery<Account> query = em.createQuery("FROM Account a WHERE a.id IN ( :accountIds )", Account.class);
        query.setParameter("accountIds", accountIds);
		return query.getResultList();
	}

	public int findByUserID(int userID) {
		Query q = em.createNativeQuery("SELECT a.id FROM accounts a LEFT JOIN users u ON u.accountID = a.id WHERE u.id = :userID");
		q.setParameter("userID", userID);

		return (Integer) q.getSingleResult();
	}
}
