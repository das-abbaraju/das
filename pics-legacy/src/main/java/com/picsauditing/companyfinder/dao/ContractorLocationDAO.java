package com.picsauditing.companyfinder.dao;

import com.picsauditing.companyfinder.model.*;
import com.picsauditing.dao.PicsDAO;
import com.picsauditing.jpa.entities.AccountStatus;
import com.picsauditing.util.Strings;
import org.apache.commons.collections.CollectionUtils;

import javax.persistence.Query;
import java.util.List;

public class ContractorLocationDAO extends PicsDAO {

    public static final String SELECT_CLAUSE = "SELECT distinct cl FROM ContractorLocation cl" +
            " JOIN cl.contractor ca";
    public static final String SELECT_CLAUSE_NATIVE =
            "SELECT distinct cl.* FROM contractor_location cl" +
                    " JOIN contractor_info ca" +
                    " ON cl.conId = ca.id" +
                    " JOIN accounts a" +
                    " ON cl.conId = a.id";

    public List<ContractorLocation> findContractorLocations(CompanyFinderFilter filter) {
        String sql = getSQL(filter);

        Query query = em.createQuery(sql);

        setQueryParameters(query, filter);

        return query.getResultList();
    }

    public List<ContractorLocationSummary> findContractorLocationsSummary(CompanyFinderFilter filter) {
        String sql = getSQLNativeSummary(filter);

        Query query = em.createNativeQuery(sql, ContractorLocationSummary.class);

        setQueryParametersSummaryNative(query, filter);

        return query.getResultList();
    }

    String getSQL(CompanyFinderFilter filter) {

        StringBuilder sql = new StringBuilder(SELECT_CLAUSE);
        List<Integer> tradeIds = filter.getTradeIds();
        if (!CollectionUtils.isEmpty(tradeIds)) {
            sql.append(" JOIN ca.trades ct");
        }

        StringBuilder whereClause = new StringBuilder(" WHERE");
        whereClause.append(getActivePendingWhere());
        whereClause.append(getViewPortWhere(filter.getViewPort()));
        whereClause.append(getTradeWhere(filter.getTradeIds()));
        whereClause.append(getSafetySensitiveWhere(filter.getSafetySensitive()));
        whereClause.append(getSoleProprietorWhere(filter.getSoleProprietor()));
        whereClause.append(getContractorIdsWhere(filter.getContractorIds()));

        sql.append(whereClause);
        return sql.toString();
    }

    String getSQLNativeSummary(CompanyFinderFilter filter) {

        StringBuilder sql = new StringBuilder(SELECT_CLAUSE_NATIVE);
        List<Integer> tradeIds = filter.getTradeIds();
        if (!CollectionUtils.isEmpty(tradeIds)) {
            sql.append(" JOIN contractor_trade ct ON a.id = ct.conId");
        }

        StringBuilder whereClause = new StringBuilder(" WHERE");
        whereClause.append(getActivePendingWhereSummaryNative());
        whereClause.append(getViewPortWhere(filter.getViewPort()));
        whereClause.append(getTradeWhereSummaryNative(filter.getTradeIds()));
        whereClause.append(getSafetySensitiveWhere(filter.getSafetySensitive()));
        whereClause.append(getSoleProprietorWhere(filter.getSoleProprietor()));
        whereClause.append(getContractorIdsWhereSummaryNative(filter.getContractorIds()));

        sql.append(whereClause);
        return sql.toString();
    }

    private String getViewPortWhere(ViewPort viewPort) {
        if (!isViewPortEmpty(viewPort)) {
            return  " AND cl.latitude > :swLat" +
                    " AND cl.longitude > :swLong" +
                    " AND cl.latitude < :neLat" +
                    " AND cl.longitude < :neLong";
        }
        return "";
    }

    private boolean isViewPortEmpty(ViewPort viewPort) {
        return viewPort == null || viewPort.isEmpty();
    }

    private String getTradeWhere(List<Integer> tradeIds) {
        if (!CollectionUtils.isEmpty(tradeIds)) {
            return " AND ct.trade.id IN :tradeList";
        }
        return "";
    }

    private String getTradeWhereSummaryNative(List<Integer> tradeIds) {
        if (!CollectionUtils.isEmpty(tradeIds)) {
            return " AND ct.id IN (:tradeList)";
        }
        return "";
    }

    private String getSafetySensitiveWhere(TriStateFlag safetySensitive) {
        if (hasSafetySensitive(safetySensitive)) {
            return " AND ca.safetySensitive = :safetySensitive";
        }
        return "";
    }

    private String getSoleProprietorWhere(TriStateFlag soleProprietor) {
        if (hasSoleProprietor(soleProprietor)) {
            return " AND ca.soleProprietor = :soleProprietor";
        }
        return "";
    }

    private String getActivePendingWhere() {
        return " (ca.status = :active OR ca.status = :pending )";
    }


    private String getActivePendingWhereSummaryNative() {
        return " (a.status = :active OR a.status = :pending )";
    }

    private String getContractorIdsWhere(List<Integer> contractorIds) {
        if (!CollectionUtils.isEmpty(contractorIds)) {
            return " AND ca.id IN :contractorIds";
        }
        return "";
    }

    private String getContractorIdsWhereSummaryNative(List<Integer> contractorIds) {
        if (!CollectionUtils.isEmpty(contractorIds)) {
            return " AND ca.id IN (:contractorIds)";
        }
        return "";
    }

    private void setQueryParameters(Query query, CompanyFinderFilter filter) {
        setViewPortQueryParams(query, filter);
        setTradeQueryParam(query, filter);
        setSoleProprietorQueryParams(query, filter);
        setSafetySensitiveQueryParam(query, filter);
        setActivePendingQueryParams(query);
        setContractorIdsQueryParam(query, filter);
    }

    private void setQueryParametersSummaryNative(Query query, CompanyFinderFilter filter) {
        setViewPortQueryParams(query, filter);
        setTradeQueryParamSummaryNative(query, filter);
        setSoleProprietorQueryParamsSummaryNative(query, filter);
        setSafetySensitiveQueryParamSummaryNative(query, filter);
        setActivePendingQueryParamsSummaryNative(query);
        setContractorIdsQueryParamSummaryNative(query, filter);
    }

    private void setViewPortQueryParams(Query query, CompanyFinderFilter filter) {
        ViewPort viewPort = filter.getViewPort();

        if(isViewPortEmpty(viewPort)){
            return;
        }

        query.setParameter("swLat", viewPort.getSouthWest().getLatitude());
        query.setParameter("swLong", viewPort.getSouthWest().getLongitude());
        query.setParameter("neLat", viewPort.getNorthEast().getLatitude());
        query.setParameter("neLong", viewPort.getNorthEast().getLongitude());
    }

    private void setTradeQueryParam(Query query, CompanyFinderFilter filter) {
        List<Integer> tradeIdList = filter.getTradeIds();

        if (CollectionUtils.isEmpty(tradeIdList)) return;

        query.setParameter("tradeList", tradeIdList);
    }

    private void setTradeQueryParamSummaryNative(Query query, CompanyFinderFilter filter) {
        List<Integer> tradeIdList = filter.getTradeIds();
        if (CollectionUtils.isEmpty(tradeIdList)) return;
        String inClauseIds = Strings.implode(tradeIdList);

        query.setParameter("tradeList", inClauseIds);
    }

    private void setSoleProprietorQueryParams(Query query, CompanyFinderFilter filter) {
        if (hasSoleProprietor(filter.getSoleProprietor())) {
            query.setParameter("soleProprietor", filter.getSoleProprietor().toBoolean());
        }
    }

    private void setSoleProprietorQueryParamsSummaryNative(Query query, CompanyFinderFilter filter) {
        if (hasSoleProprietor(filter.getSoleProprietor())) {
            query.setParameter("soleProprietor", filter.getSoleProprietor().getValue());
        }
    }

    private void setSafetySensitiveQueryParam(Query query, CompanyFinderFilter filter) {
        if (hasSafetySensitive(filter.getSafetySensitive())) {
            query.setParameter("safetySensitive", filter.getSafetySensitive().toBoolean());
        }
    }

    private void setSafetySensitiveQueryParamSummaryNative(Query query, CompanyFinderFilter filter) {
        if (hasSafetySensitive(filter.getSafetySensitive())) {
            query.setParameter("safetySensitive", filter.getSafetySensitive().getValue());
        }
    }

    private void setActivePendingQueryParams(Query query) {
        query.setParameter("active", AccountStatus.Active);
        query.setParameter("pending", AccountStatus.Pending);
    }

    private void setActivePendingQueryParamsSummaryNative(Query query) {
        query.setParameter("active", AccountStatus.Active.name());
        query.setParameter("pending", AccountStatus.Pending.name());
    }

    private void setContractorIdsQueryParam(Query query, CompanyFinderFilter filter) {
        List<Integer> contractorIdList = filter.getContractorIds();
        if (CollectionUtils.isEmpty(contractorIdList)) return;
        query.setParameter("contractorIds", contractorIdList);
    }

    private void setContractorIdsQueryParamSummaryNative(Query query, CompanyFinderFilter filter) {
        List<Integer> contractorIdList = filter.getContractorIds();
        if (CollectionUtils.isEmpty(contractorIdList)) return;
        String inClauseIds = Strings.implode(contractorIdList);

        query.setParameter("contractorIds", inClauseIds);
    }

    private boolean hasSafetySensitive(TriStateFlag safetySensitive) {
        return safetySensitive != TriStateFlag.IGNORE;
    }

    private boolean hasSoleProprietor(TriStateFlag soleProprietor) {
        return soleProprietor != TriStateFlag.IGNORE;
    }

    public ContractorLocation findById(int conId) {
        String sql = "SELECT cl FROM ContractorLocation cl WHERE cl.contractor.id = :conId";
        Query query = em.createQuery(sql);
        query.setParameter("conId", conId);
        List<ContractorLocation> contractorLocation = query.getResultList();
        if (!contractorLocation.isEmpty()) {
            return contractorLocation.get(0);
        }
        return null;
    }
}
