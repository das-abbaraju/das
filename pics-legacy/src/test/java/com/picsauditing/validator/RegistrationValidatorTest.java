package com.picsauditing.validator;

import static org.mockito.Matchers.anyString;
import static org.mockito.Mockito.when;

import java.util.Collections;
import java.util.List;
import java.util.Map;
import java.util.TimeZone;

import com.picsauditing.PicsTestUtil;
import com.picsauditing.model.i18n.LanguageModel;
import edu.emory.mathcs.backport.java.util.Arrays;
import org.approvaltests.Approvals;
import org.approvaltests.reporters.DiffReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.Before;
import org.junit.Test;
import org.mockito.*;

import com.picsauditing.dao.ContractorAccountDAO;
import com.picsauditing.jpa.entities.ContractorAccount;
import com.picsauditing.jpa.entities.Country;

@UseReporter(DiffReporter.class)
public class RegistrationValidatorTest {

    private VATValidator vatValidator;

    @Mock
    private VATWebValidator vatWebValidator;

    @Before
    public void setup() throws Exception {
        MockitoAnnotations.initMocks(this);

        vatValidator = new VATValidator();
        PicsTestUtil.forceSetPrivateField(vatValidator, "webValidator", vatWebValidator);
    }

	@Test
	public void testValidateContractor_NoValues_PopulatesErrors() throws Exception {
		String language = null;
		String dialect = null;
		ContractorAccount contractor = new ContractorAccount();
		contractor.setCountry(new Country("en_US"));
		String countrySubdivision = null;

        when(vatWebValidator.execute()).thenReturn(Boolean.FALSE);

		verifyValidation(language, dialect, contractor, countrySubdivision, Arrays.asList(new Country[] { new Country("US") }));
	}

	@Test
	public void testValidateContractor_ValidSwedish_ErrorsWillBeEmpty() throws Exception {
		String language = "sv";
		String dialect = null;
		ContractorAccount contractor = new ContractorAccount();
		contractor.setName("Sven Nilsson");
		contractor.setAddress("Roslagsgatan 10");
		contractor.setCity("Stockholm");
		contractor.setVatId("SE989898999901");
		contractor.setCountry(new Country("SE"));
		contractor.setZip("113 51");
		contractor.setTimezone(TimeZone.getDefault());
		String countrySubdivision = null;

        when(vatWebValidator.execute()).thenReturn(Boolean.TRUE);

		verifyValidation(language, dialect, contractor, countrySubdivision, Collections.<Country>emptyList());
	}

	private void verifyValidation(String language, String dialect, ContractorAccount contractor,
								  String countrySubdivision, List<Country> dialects) throws Exception {
		ContractorAccountDAO contractorAccountDao = Mockito.mock(ContractorAccountDAO.class);
		when(contractorAccountDao.findByCompanyName(anyString())).thenReturn(null);
		LanguageModel languageModel = Mockito.mock(LanguageModel.class);
		when(languageModel.getDialectCountriesBasedOn(anyString())).thenReturn(dialects);

		Map<String, String> errors = RegistrationValidator.validateContractor(language, dialect, contractor,
				countrySubdivision, languageModel, new InputValidator().setContractorAccountDao(contractorAccountDao),
				vatValidator);
		Approvals.verify(errors);
	}
}
