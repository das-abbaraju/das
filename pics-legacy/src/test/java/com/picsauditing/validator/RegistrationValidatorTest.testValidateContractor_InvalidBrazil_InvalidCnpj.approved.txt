contractor.vatId : JS.Validation.InvalidTaxID::CNPJ 
